src/network/clientpackethandler.cpp

```diff
@@ -1433,3 +1434,35 @@ void Client::handleCommand_ModChannelSignal(NetworkPacket *pkt)
 	if (valid_signal)
 		m_script->on_modchannel_signal(channel, signal);
 }
+
+void Client::handleCommand_ParticleOverlay(NetworkPacket *pkt)
+{
+	std::string specName;
+	// Load spec name first to find the clientEnvironment spec associated
+	*pkt >> specName;
+
+	ParticleOverlaySpec *spec = m_env.getParticleSpecOverlay(specName);
+	spec->name = specName;
+	*pkt >> spec->texture_name >> spec->texture_scale_factor.Width
+		>> spec->texture_scale_factor.Height >> spec->minpps >> spec->maxpps
+		>> spec->direction >> spec->velocity >> spec->gravity_factor >> spec->enabled;
+
+	// Limit max PPS to prevent bad performance values
+#ifdef __ANDROID__
+	// Android devices are less powerful, lower the limits
+	if (spec->minpps > 5000)
```

Comment:
`spec->minpp = rangelim(spec->minpps, 0, 5000);` to ensure that all values are in the range we want them (even though it's sent by the server)